src/nwp_consumer/internal/inputs/cmc/client.py

```diff
@@ -0,0 +1,326 @@
+import bz2
+import datetime as dt
+import pathlib
+import re
+import typing
+import urllib.request
+
+import requests
+import structlog
+import xarray as xr
+
+from nwp_consumer import internal
+
+from ._consts import GDPS_VARIABLES, GEPS_VARIABLES
+from ._models import CMCFileInfo
+
+log = structlog.getLogger()
+
+# See https://eccc-msc.github.io/open-data/msc-data/nwp_gdps/readme_gdps-datamart_en/ for a list of CMC parameters
+PARAMETER_RENAME_MAP: dict[str, str] = {
+    "t": internal.OCFShortName.TemperatureAGL.value,
+    "tclc": internal.OCFShortName.LowCloudCover.value, # TODO: Check this is okay
+    "dswrf": internal.OCFShortName.DownwardShortWaveRadiationFlux.value,
+    "dlwrf": internal.OCFShortName.DownwardLongWaveRadiationFlux.value,
+    "snod": internal.OCFShortName.SnowDepthWaterEquivalent.value,
+    "rh": internal.OCFShortName.RelativeHumidityAGL.value,
+    "u": internal.OCFShortName.WindUComponentAGL.value,
+    "v": internal.OCFShortName.WindVComponentAGL.value,
+}
+
+COORDINATE_ALLOW_LIST: typing.Sequence[str] = ("time", "step", "latitude", "longitude")
+
+
+class Client(internal.FetcherInterface):
+    """Implements a client to fetch ICON data from DWD."""
+
+    baseurl: str  # The base URL for the ICON model
+    model: str  # The model to fetch data for
+    parameters: list[str]  # The parameters to fetch
+    conform: bool  # Whether to rename parameters to OCF names and clear unwanted coordinates
+
+    def __init__(self, model: str, hours: int = 48, param_group: str = "default") -> None:
+        """Create a new GDPS Client.
+
+        Exposes a client for GDPS and GEPS data from Canada CMC that conforms to the FetcherInterface.
+
+        Args:
+            model: The model to fetch data for. Valid models are "gdps" and "geps".
+            param_group: The set of parameters to fetch.
+                Valid groups are "default", "full", and "basic".
+        """
+        self.baseurl = "https://dd.weather.gc.ca"
+
+        match model:
+            case "gdps":
+                self.baseurl += "/model_gem_global/15km/grib2/lat_lon/"
+            case "geps":
+                self.baseurl += "/ensemble/geps/grib2/raw/"
+            case _:
+                raise ValueError(
+                    f"unknown icon model {model}. Valid models are 'gdps' and 'geps'",
+                )
+
+        match (param_group, model):
+            case ("default", _):
+                self.parameters = list(PARAMETER_RENAME_MAP.keys())
+                self.conform = True
+            case ("full", "geps"):
+                self.parameters = GEPS_VARIABLES
+                self.conform = False
+            case ("full", "gdps"):
+                self.parameters = GDPS_VARIABLES
+                self.conform = False
+            case (_, _):
+                raise ValueError(
+                    f"unknown parameter group {param_group}."
+                    "Valid groups are 'default', 'full', 'basic'",
```

Comment:
Doesn't look like there's a basic implemented - but I can do that along with an integration test if you like!

Reply:
Yeah, I guess not sure what basic would cover that isn't the default, or full one? The default seems like the basic one to me

Reply:
Basic is built for the purpose of quickening the integration tests, so it would be a very small subset of the available parameters - two of them say - enought to check it can still merge datasets correctly but no more.